src/PrincipalToken.sol

```diff
@@ -1,86 +1,82 @@
 // SPDX-License-Identifier: CC0-1.0
-pragma solidity ^0.8.20;
+pragma solidity 0.8.20;
```

Comment:
Let's keep the floating pragma, makes cross-compatibility and imports easier. We'll deploy with a specific version but doesn't need to be that tightly constrained in the repo. Also applies to the other changes at the top of other files

Reply:
Think I see what you're saying. Certainly, for interfaces and libraries that could be imported by other projects, makes sense to allow higher compiler versions. But for core contracts it is definitely best practice to have them fixed so the bytecode is consistent across testing. So, maybe in this repo, `PrincipalToken.sol`, `WrapOfferer.sol` and `DelegateToken.sol` fixed and floating for the interfaces? Think that addresses the cross-compatibility / imports issue without the issues of floating pragma for the core contracts.

Reply:
We can just set the compiler version in foundry.toml when we need it to be consistent for testing. Think it's cleanest, say somebody copy-pastes a file into another repo they're using, to not have to have them manually modify pragmas. Don't think we need to spend much time or energy at all worrying about compiler version differences, can just assume accurate behavior and forwards compatibility

Reply:
Let's stick with the floating pragma for simpler devex on integrators running local tests. Understood this introduces the risk of forward compatibility if they sim on a later version without doing mainnet fork testing as well.

Reply:
Set to ^0.8.0 for interfaces, ^0.8.21 otherwise

Reply:
Bumped interfaces to ^0.8.4 to avoid compiler errors given custom errors